static/js/protein/data_processing_utils.ts

```diff
@@ -592,3 +529,248 @@ export function getProteinDescription(data: GraphNodes): string {
+  // checks if the data is empty or not
+  if (_.isEmpty(data)) {
+    return;
+  }
+
+  // P53_HUMAN is central protein in below examples.
+  // take interaction names of the form P53_HUMAN_ASPP2_HUMAN | ASPP2_HUMAN_P53_HUMAN and parse into ASPP2_HUMAN.
+  const centerNodeId = data[0].parent;
+  let neighbors = data.map(({ name: interactionId, value }) => {
+    // value is confidenceScore
+    const neighborId = getInteractionTarget(
+      ppiDcidFromId(interactionId),
+      ppiDcidFromId(centerNodeId)
+    );
+    const nodeDatum = nodeFromId(neighborId, 1);
+    nodeDatum["value"] = value;
+    return nodeDatum;
+  });
+
+  // delete duplicates and self-interactions (will add support for self-interactions later on)
+  const seen = new Set();
+  neighbors = neighbors.filter((node) => {
+    const duplicate = seen.has(node.name);
+    seen.add(node.name);
+    return !duplicate && node.id !== centerNodeId;
+  });
+
+  // descending order of interaction confidenceScore
+  neighbors.sort((n1, n2) => n2.value - n1.value);
+  // consider only top 10 interactions to avoid clutter
+  neighbors = neighbors.slice(0, MAX_INTERACTIONS);
+
+  const centerDatum = nodeFromId(centerNodeId, 0);
+
+  const linkData: InteractionLink[] = neighbors.map((node) => {
+    return {
+      score: node.value,
+      source: centerNodeId,
+      target: node.id,
+    };
+  });
+
+  return {
+    // guaranteed to be [] because we don't support self-interactions yet
+    linkDataNested: [[], linkData],
+    nodeDataNested: [[centerDatum], neighbors],
+  };
+}
+
+/**
+ * Given response and key, map each response datum to value of key and return map
+ */
+export function getFromResponse<K extends keyof V1BioResponseDatum>(
+  resp: V1BioResponse,
+  key: K
+): V1BioResponseDatum[K][] {
+  if (!("data" in resp.data)) {
+    return [];
+  }
+  return resp.data.map((obj) => obj[key]);
+}
+
+/**
+ * Given an array of interaction DCIDs and a parallel array of corresponding scores,
+ * construct and return score record such that for each interaction A_B with corresponding DCID in interaction DCIDs,
+ * both A_B and B_A map to the score of A_B.
+ *
+ * Since we currently do not graphically distinguish between the A_B and B_A scores,
+ * we choose to have a symmetric score store.
+ *
+ * In the future perhaps we should make the graph directed to make relationships like "A phosphorylates B"
+ * more clear. I also think there should be assays for which the scores are not symmetric -
+ * ex. "knockout A and see what happens to the expression of B". These might not exist in the graph currently
+ * but could very well be imported in the future. Thus, we might have two or more edges between proteins
+ * for each type of interaction they participate in, with different scores for each.
+ */
+export function symmetricScoreRec(
+  interactionDcids: bioDcid[],
+  scores: number[]
+): Record<string, number> {
+  const scoreRec = {};
+  for (let i = 0; i < interactionDcids.length; i++) {
```

Comment:
in case the two arrays are different lengths, you want the termination condition to be `i < Math.min(interactionDcids.length, scores.length)`

Reply:
ah ok that's right, thanks for catching!